docs/pages/Architecture/Discovery and Peering/discovery-component.md

```diff
@@ -3,7 +3,30 @@ title: "Discovery Protocol"
 ---
 
 ## Overview
-This component's goal is to find other clusters running Liqo around us, get information needed to pair and start peering process
+The goal of this component is to find other clusters running Liqo on the same LAN, and to obtain the information required to possibly start the peering process.
+
+
+## Peering Process
+
+1. When the `Join` flag in the `ForeignCluster` CR becomes true (either automatically or manually), an operator is triggered that creates a new `FederationRequest` CR in the _foreign cluster_.
+The `FederationRequest` creation process includes the creation of new `kubeconfig` with management permissions on `Advertisement` CRs. It is worth noting that, by default, each Liqo cluster grants the _create-only_ permission on `FederationRequest` resources to unauthenticated user.
+2. In the foreign cluster, an admission webhook accepts/rejects `FederationRequest`s.
+3. The information contained in the `FederationRequest` is used to start the sharing of resources.
+
+
+## Neighbor discovery
+
+Discovery service allows two clusters to know each other, ask for resources and begin exchanging `Advertisements`.
+The protocol is described by the following steps:
+
+1. Each cluster registers its master IP, its ClusterID and the namespace where Liqo is deployed to a mDNS service
+2. The requesting cluster sends a mDNS query on the local network to find available foreigns clusters
+3. When a reply is received, the requesting cluster reads the data associated to the foreign cluster (i.e., ClusterID, etc) in the mDNS reply
+4. The home cluster stores all this information in a `ForeignCluster` CR
+
+mDNS packets are very similar to the ones exchanged in DNS discovery with exception of PTR records.
+In mDNS discovery list of all clusters will be the ones that replies on multicast query on `local.` domain.
```

Comment:
@aleoli Please check this sentence. Really unable to understand what this means.